### DISCUSSION TITLE
compress error when importing file into adobe premiere

### DISCUSSION THREAD

**Discussion by pierregit1** [streamlink/streamlink#4449] [Q&A] [answered]
hi,
first of all, i'm very new to all this and i'm definitely not an user of python etc.
i'm having absolutely 0 issue about using streamlink to watch a stream or keep a vod of it when i'm not home to watch.
my only issue is: i can't import video files into adobe premiere and i work as a video editor for a streamer, which is a bit annoying. i can't get my hands on the vods often because of the DMCA stuff so I like the idea of recording them myself. 
whenever i want to import a vod into adobe premiere, i get the error saying the compress type isnt supported by adobe premiere and i'm having a really hard time finding how to change it
I tried converting with ffmpeg, which work once its done but i've to do it manually everytime and it take a very long time for vods of 4 hours and more.
is there a way for me to work around it and make the vod directly into a compatible file for adobe premiere ?
thanks for reading

**Comment by bastimeyer** [comment, marked as answer]
> i work as a video editor for a streamer
> is there a way for me to work around it and make the vod directly into a compatible file for adobe premiere ?

What is "the VOD"? Are you talking about Twitch?
Twitch streams are HLS streams with MPEG-TS transport streams consisting of h264 video and aac audio streams.

All you need to do is remux it to the MPEG-4 container in order to fix the timestamps and make it compatible with your video editor. No re-encoding is required.

```sh
ffmpeg -i "$input.ts" -c copy "$output.mp4"
```

This can even be done on-the-fly, which has been answered on the issue tracker several times in the past:

```sh
streamlink --stdout "$url" "$quality" | ffmpeg -i pipe:0 -c copy "$output.mp4"
```

- https://streamlink.github.io/latest/cli.html#cmdoption-stdout
- https://ffmpeg.org/ffmpeg-protocols.html#pipe

**Reply by pierregit1** [reply]
I'm sorry if this has been asked in the past, I've tried to look before asking but since I've pretty much no idea what I was looking for, it was a bit hard.
Thank you for your answer. If I understand, all I've to do is use the this command once the stream start
`streamlink --stdout "$url" "$quality" | ffmpeg -i pipe:0 -c copy "$output.mp4"`
and I'll keep a video of the stream once it's over and it will work ?
Thank you for your answer

**Reply by bastimeyer** [reply]
> all I've to do is use the this command

All you have to do is remux from MPEG-TS to MPEG-4. How you do this is up to you. Don't run any commands posted on the internet which you don't know what they do.
